### DISCUSSION TITLE
kitten ssh remote host with read only homedir

### DISCUSSION THREAD

**Discussion by ragazenta** [kovidgoyal/kitty#7778] [Q&A] [answered]
```
mktemp: failed to create directory via template ‘/home/user/.kitty-ssh-kitten-untar-XXXXXXXXXXXX’: Read-only file system
Creating temp directory failed
```

Is there any workaround to support this without installing `kitty-terminfo`.

Thanks.

**Comment by kovidgoyal** [comment, marked as answer]
No, there isn't.

**Reply by ragazenta** [reply]
I'm curious about this line, I found it inside `/usr/lib64/kitty`.

https://github.com/kovidgoyal/kitty/blob/9843b5c2108cd883285f2f0944637effb26d90c0/shell-integration/ssh/bootstrap.sh#L108

I've modified it to `mktemp`  on different path, but kitten ssh still use the same path.

**Reply by kovidgoyal** [reply]
If you want to modify kitty source, you need to build it as well. https://sw.kovidgoyal.net/kitty/build/

**Reply by ragazenta** [reply]
Oh, I thought kitty will just call the modified shell script. Then what is the purpose of those scripts provided in the package (rpm or deb), if I cannot run modified one without rebuilding?

**Reply by adamcstephens** [reply]
Would you consider a flag for this? I provision the remote with nix and the terminfo is already installed but read-only due to the nix store.

```
❯ eza -lg ~/.terminfo/x/xterm-kitty 
.r--r--r-- 3.7k root root 31 Dec  1969 /home/adam/.terminfo/x/xterm-kitty
```

**Reply by kovidgoyal** [reply]
A flag for what exactly? If you want an option to disable copying the terminfo into ~/.terminfo, sure that can be done, it should be a simple PR.

**Reply by adamcstephens** [reply]
> If you want an option to disable copying the terminfo into ~/.terminfo

Sorry, yes this is what I was asking for.

**Reply by kovidgoyal** [reply]
Sure, as I said simple PR. Or if you cant figure it out open an issue for it and I will look at it when I have the time.
